src/Componets/SelectCategory/__test__/SelectCattegory.test.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import {
+  fireEvent,
+  render, screen,
+} from '@testing-library/react';
+import Select from '../SelectCategory';
+
+describe('NameInput', () => {
+  it('should render and have placeholder', () => {
+    render(<Select
+      value=""
+      onChange={() => {}}
+      selected={[]}
+      options={['dog', 'cats']}
+      placeholder="Categories"
+    />);
+
+    const inputElement = screen.getByText('Categories');
+    expect(inputElement).toBeInTheDocument();
+  });
+  it('should have options to select', () => {
+    render(<Select
+      value=""
+      onChange={() => {}}
+      selected={[]}
+      options={['dog', 'cats']}
+      placeholder="Categories"
+    />);
+
+    const optionElements = screen.getAllByTestId('select-category-options');
+    optionElements.forEach((optionElement) => {
+      expect(optionElement).toBeInTheDocument();
+    });
```

Comment:
if for some reason `optionElements` are not found (they are an empty array) this test will not fail

Reply:
It might be better to check if optionElements are really what they should be:
Check if they have the same length as expected:
`expect(optionElement.length).toBeEqual(mockedOptions.lenght);`
and then check if they have the same text as expected (not sure about syntax but you should get the point):
```
optionElements.forEach((optionElement, index) => {
      expect(optionElement). toHaveTextContent(mockedOptions[index);
    });
```